myriadeploy/setup_cluster.py

```diff
@@ -2,55 +2,30 @@
 
 import subprocess
 import sys
-import remove_deployment
 import logging
 
-
-def make_catalog(config_file):
-    """Creates a Myria catalog (running the Java program to do so) from the
-given deployment configuration."""
-    args = ["./run_catalog_maker.sh", config_file]
-    if subprocess.call(args):
-        logging.error("Error making the catalogs")
-
-
-def copy_catalogs(config_file):
-    """Copies the master and worker catalogs to the remote hosts."""
-    args = ["./using_deployment_utils.sh", config_file, "-copy_master_catalog"]
-    if subprocess.call(args):
-        logging.error("Error copying master catalog")
-    args = ["./using_deployment_utils.sh", config_file, "-copy_worker_catalogs"]
-    if subprocess.call(args):
-        logging.error("Error copying worker catalogs")
-
-
-def copy_distribution(config_file):
+def make_deployment(config_file, fresh):
     "Copy the distribution (jar and libs and conf) to compute nodes."
-    args = ["./using_deployment_utils.sh", config_file, "-copy_distribution"]
+    args = ["./using_deployment_utils.sh", config_file, "-deploy"]
+    if fresh:
+        args.append("-fresh_catalog")
     if subprocess.call(args):
         logging.error("Error copying distribution")
 
 
 def main(argv):
     # Usage
-    if len(argv) != 2:
-        print >> sys.stderr, "Usage: %s <deployment.cfg>" % (argv[0])
+    if len(argv) < 2 or len(argv) > 3 or (len(argv) == 3 and argv[2] != "-fresh_catalog"):
+        print >> sys.stderr, "Usage: %s <deployment.cfg> <optional: -fresh_catalog>" % (argv[0])
```

Comment:
In most Unix programs, options precede ordinary arguments and (outside Java-land) are in short form (one letter, with a single-hyphen prefix) or long form (one or more hyphen-separated complete words, with a two-hyphen prefix).